phys2bids/physio_obj.py

```diff
@@ -228,19 +401,22 @@ def delete_at_index(self, idx):
             Removes element at index idx
         self.units:
             Removes element at index idx
-        self.ch_amount:
-            In all the property that are lists, the element correspondent to
-            `idx` gets deleted
+        self.trigger_idx:
+            If the deleted index was the one of the trigger, set the trigger idx to 0.
         """
         del self.timeseries[idx]
         del self.freq[idx]
         del self.ch_name[idx]
         del self.units[idx]
 
-    def check_trigger_amount(self, chtrig=1, thr=None, num_timepoints_expected=0, tr=0):
+        if self.trigger_idx == idx:
+            LGR.warning('Removing trigger channel - are you sure you are doing'
+                        'the right thing?')
```

Comment:
Maybe we should wait for the user's input at this point and only proceed if the user confirms.

Reply:
That could be an option, but then if you run a batch of files they won't run smoothly - if there's a reason for the user to remove the trigger channel.
Why don't we add the possibility of such interaction in a different PR? We can add a parser argument that would pause the workflow in particular cases.

Reply:
That's what I was thinking actually; i.e. to add an argument to the parser. However, I'd do it so that it stops by default. It would be like giving the "pro" option to power-users.

What do you think?

Reply:
Sure, agreed on having an option! Can you open an enhancement issue about it?
Don't mark it for the beta milestone though - we have enough stuff in it.